pkg/naming/port.go

```diff
@@ -0,0 +1,30 @@
+package naming
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	// DNS_LABEL constraints: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+	dnsLabelValidation = regexp.MustCompile("^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$")
+)
+
+// PortName defines the port name used in services, ingresses and routes.
+// The port name in pods is validated differently it has to be 15 characters long.
+func PortName(receiverName string, port int32) string {
```

Comment:
This was moved from `parser`  package. It is being called from multiple places now